File: web/src/api/ConsultingAPI/consultingAPI.js
import { baseAxios } from '../Axios.jsx';

//상담 예정 목록
export const getConsultingList = async () => {
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');
  try {
    const response = await baseAxios.get(
      `/consult-service/v1/web/${schoolYear}/schools/${schoolNum}/consults?status=5001`,
      {
        headers: {
          Authorization: `Bearer ${sessionStorage.getItem('token')}`
        }
      }
    );
    return response.data.data.content;
  } catch (error) {
    return 0;
  }
};
//상담 승인
export const approveConsulting = async (consultId) => {
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');

  try {
    const response = await baseAxios.patch(
      `/consult-service/v1/web/${schoolYear}/schools/${schoolNum}/consults/${consultId}/approve`
    );
    console.log(response.data);
    if (response.data.message === 'SUCCESS') {
      alert('상담 승인이 완료되었습니다.');
      window.location.reload();
    }
    return response.data;
  } catch (error) {
    return 0;
  }
};
//상담 거절
export const rejectConsulting = async (consultId, reason) => {
  const rejectedReason = {
    rejectedReason: reason
  };
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');

  try {
    const response = await baseAxios.patch(
      `/consult-service/v1/web/${schoolYear}/schools/${schoolNum}/consults/${consultId}/reject`,
      rejectedReason
    );
    console.log(response.data);
    if (response.data.message === 'SUCCESS') {
      alert('거절되었습니다.');
      window.location.reload();
    }
    return response.data;
  } catch (error) {
    return 0;
  }
};

export const getConsultingMessage = async () => {
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');
  try {
    const response = await baseAxios.get(
      `/consult-service/v1/app/${schoolYear}/schools/${schoolNum}/consult-schedules`,
      {
        headers: {
          Authorization: `Bearer ${sessionStorage.getItem('token')}`
        }
      }
    );
    console.log(response.data.data);
    return response.data.data;
  } catch (error) {
    return 0;
  }
};

export const modifyConsultMsg = async (consultScheduleId, description) => {
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');
  const info = {
    description: description
  };

  try {
    const response = await baseAxios.patch(
      `/consult-service/v1/app/${schoolYear}/schools/${schoolNum}/consult-schedules/${consultScheduleId}/description`,
      info,
      {
        headers: {
          Authorization: `Bearer ${sessionStorage.getItem('token')}`
        }
      }
    );
    return response.data.data;
  } catch (error) {
    return 0;
  }
};

export const modifyConsultTime = async (consultScheduleId, times) => {
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');
  console.log(times.monday);
  const info = {
    monday: times.monday,
    tuesday: times.tuesday,
    wednesday: times.wednesday,
    thursday: times.thursday,
    friday: times.friday
  };

  try {
    const response = await baseAxios.patch(
      `/consult-service/v1/app/${schoolYear}/schools/${schoolNum}/consult-schedules/${consultScheduleId}/schedules`,
      info,
      {
        headers: {
          Authorization: `Bearer ${sessionStorage.getItem('token')}`
        }
      }
    );
    console.log(response.data);
    return response.data.data;
  } catch (error) {
    return 0;
  }
};

//상담 내역
export const getConsults = async (status) => {
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');

  try {
    const response = await baseAxios.get(
      `/consult-service/v1/web/${schoolYear}/schools/${schoolNum}/consults?status=${status}`,
      {
        headers: {
          Authorization: `Bearer ${sessionStorage.getItem('token')}`
        }
      }
    );

    return response.data.data;
  } catch (error) {
    return 0;
  }
};

//상담 내역
export const getCompliteConsulting = async () => {
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');

  try {
    const response = await baseAxios.get(
      `/consult-service/v1/web/${schoolYear}/schools/${schoolNum}/consults?status=5001`,
      {
        headers: {
          Authorization: `Bearer ${sessionStorage.getItem('token')}`
        }
      }
    );

    return response.data.data;
  } catch (error) {
    return 0;
  }
};

//상담 상세 내역
export const getConsultDetail = async (consultId) => {
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');

  try {
    const response = await baseAxios.get(
      `/consult-service/v1/web/${schoolYear}/schools/${schoolNum}/consults/${consultId}`,
      {
        headers: {
          Authorization: `Bearer ${sessionStorage.getItem('token')}`
        }
      }
    );
    console.log(response);
    return response.data.data;
  } catch (error) {
    return 0;
  }
};

export const sendCompliteConsult = async (consultId, result) => {
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');

  try {
    const response = await baseAxios.patch(
      `/consult-service/v1/web/${schoolYear}/schools/${schoolNum}/consults/${consultId}/finish`,
      { resultContent: result },
      {
        headers: {
          Authorization: `Bearer ${sessionStorage.getItem('token')}`
        }
      }
    );
    console.log(response);
    return response.data;
  } catch (error) {
    return 0;
  }
};

export const getConsultSchedule = async (consultId, result) => {
  const schoolNum = sessionStorage.getItem('schoolId');
  const schoolYear = sessionStorage.getItem('year');

  try {
    const response = await baseAxios.patch(
      `/consult-service/v1/app/${schoolYear}/schools/${schoolNum}/consult-schedules/`,
      { resultContent: result },
      {
        headers: {
          Authorization: `Bearer ${sessionStorage.getItem('token')}`
        }
      }
    );
    console.log(response);
    return response.data;
  } catch (error) {
    return 0;
  }
};

File: app/everyschool/lib/api/consulting_api.dart
import 'package:dio/dio.dart';
import 'package:everyschool/api/base_api.dart';
import 'package:flutter_secure_storage/flutter_secure_storage.dart';

class ConsultingApi {
  final storage = FlutterSecureStorage();
  ServerApi serverApi = ServerApi();
  Dio dio = Dio();

  Future<dynamic> getConsultingList(schoolId, year) async {
    var userType = await storage.read(key: 'usertype');
    String? token = await storage.read(key: 'token');

    String lastUrl;

    if (userType == '1002') {
      lastUrl =
          '/consult-service/v1/app/$year/schools/$schoolId/consults/parent';
    } else {
      lastUrl =
          '/consult-service/v1/app/$year/schools/$schoolId/consults/teacher';
    }

    try {
      final response = await dio.get('${serverApi.serverURL}$lastUrl',
          options: Options(headers: {'Authorization': 'Bearer $token'}));
      print('주소 ${serverApi.serverURL}$lastUrl');
      return response.data['data'];
    } catch (e) {
      print(e);
      return 0;
    }
  }

  Future<dynamic> getConsultDetail(schoolId, year, consultId) async {
    var userType = await storage.read(key: 'usertype');
    String? token = await storage.read(key: 'token');

    String lastUrl;

    if (userType == '1002') {
      lastUrl =
          '/consult-service/v1/app/$year/schools/$schoolId/consults/parent/$consultId';
    } else {
      lastUrl =
          '/consult-service/v1/app/$year/schools/$schoolId/consults/teacher/$consultId';
    }

    try {
      final response = await dio.get('${serverApi.serverURL}$lastUrl',
          options: Options(headers: {'Authorization': 'Bearer $token'}));
      return response.data['data'];
    } catch (e) {
      print(e);
      return 0;
    }
  }

  Future<dynamic> getTeacherId(year, schoolId) async {
    String? token = await storage.read(key: 'token');
    try {
      final response = await dio.get(
          '${serverApi.serverURL}/user-service/v1/app/$year/schools/$schoolId/students',
          options: Options(headers: {'Authorization': 'Bearer $token'}));
      return response.data['data'];
    } catch (e) {
      print(e);
    }
  }

  Future<dynamic> getConsultSchedule(
      schoolId, year, consultId, teacherId) async {
    String? token = await storage.read(key: 'token');

    try {
      final response = await dio.get(
          '${serverApi.serverURL}/consult-service/v1/app/$year/schools/$schoolId/consult-schedules/$teacherId',
          options: Options(headers: {'Authorization': 'Bearer $token'}));

      return response.data['data'];
    } catch (e) {
      print(e);
      return 0;
    }
  }

  Future<dynamic> registerConsult(year, schoolId, teacherKey, studentKey,
      typeId, consultDateTime, message) async {
    String? token = await storage.read(key: 'token');
    var data = {
      "teacherKey": teacherKey,
      "studentKey": studentKey,
      "typeId": typeId,
      "consultDateTime": consultDateTime,
      "message": message
    };

    try {
      final response = await dio.post(
          '${serverApi.serverURL}/consult-service/v1/app/$year/schools/$schoolId/consults',
          data: data,
          options: Options(headers: {'Authorization': 'Bearer $token'}));

      print(response.data);

      return response.data;
    } catch (e) {
      print(e);
      return 0;
    }
  }
}
